### PR TITLE
Feature modify board sim

### PR DESCRIPTION
**Pull request by kigateG**
Changed board and added dxl, ir drivers for simulator.

### CODE REVIEW

File: libs/irSensor/docs/reference/input.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+```cards
+input.onTemperatureConditionChanged(TemperatureCondition.Hot, 15, TemperatureUnit.Celsius, () => {
+	
+})
+input.temperature(TemperatureUnit.Celsius);
+```
+
+```package
+thermometer
```
Comment (pelikhan) [inline libs/irSensor/docs/reference/input.md]: Irsensor?

File: libs/dynamixel/sim/dynamixel.ts [hunk 2]
```diff
@@ -30,3 +30,13 @@ namespace pxsim.DynamixelDeviceMethods {
         device.write(id, addr, buffer);
     }
 }
+
+namespace pxsim.dxl{
+    export function setPosition(device:DynamixelDevice, id:number, position:number):void{
```
Comment (pelikhan) [inline libs/dynamixel/sim/dynamixel.ts]: Needs to match the shins bame. dxl-> dynamixel ?

File: libs/dynamixel/sim/state.ts [hunk 3]
```diff
@@ -52,5 +52,21 @@ namespace pxsim {
         write(id: number, addr: number, buffer: RefBuffer): void {
             // TODO
         }
+
+        /**
+         */
+        //% shim=dxl::__setPosition
```
Comment (pelikhan) [inline libs/dynamixel/sim/state.ts]: This shims won’t work on hardware games they are JavaScript only

File: libs/irSensor/jacdac.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/// <reference path="../../node_modules/pxt-common-packages/libs/jacdac/sensordriver.ts"/>
+/// <reference path="../../node_modules/pxt-common-packages/libs/jacdac/config.ts"/>
```
Comment (pelikhan) [inline libs/irSensor/jacdac.ts]: Don’t worry about Jacdac for now.

File: libs/irSensor/pxt.json [hunk 5]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "irSensor",
```
Comment (pelikhan) [inline libs/irSensor/pxt.json]: We keep all package names lower case

File: libs/irSensor/sim/state.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    export interface IrSensorBoard extends CommonBoard {
+        irSensorState: IrSensorState;
+    }
+
+    export function irSensorState(): IrSensorState {
+        return (board() as IrSensorBoard).irSensorState;
+    }
+
+    export class IrSensorState {
+        public sensorUsed: boolean = false;
```
Comment (pelikhan) [inline libs/irSensor/sim/state.ts]: You need to  by use a level detector to generate the hot/cold events

File: libs/lightRGB/lightRGB.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+    }
+
+    /**
+     * Converts wheel position into an RGB color
+     * @param wheelPos value between 0 to 255 to get a color value, eg: 99
+     */
+    //% blockId="rgb_wheel" block="color slider %wheelPos=colorWheelPicker"
+    //% group="Colors"
+    //% weight=19 help="rgb/wheel" blockGap=8
+    export function wheel(wheelPos: number): number {
```
Comment (pelikhan) [inline libs/lightRGB/lightRGB.ts:91]: Use color package if possible

**Review by pelikhan — APPROVED** [review]
This should have been multiple PR. I am not going to review massive PRs.

I assume you tested this in the simulator and on hardware.

**Comment by kigateG** [comment]
Thank you Peli for the comments. We will fix it.

File: libs/irSensor/sim/state.ts (hunk 6 above)
Comment (HgonKim) [reply libs/irSensor/sim/state.ts]: This class is just test for **Ir Sensor** simulation.
I'm not sure whether to generate the hot/cold events.

**Pull request closed without merging**